Frameworks/QuartzCore/CALayer.mm

```diff
@@ -353,19 +366,28 @@ void ReleaseDisplayTexture(DisplayTexture* tex) {
 static LockingBufferInterface _globallockingBufferInterface;
 
 CGContextRef CreateLayerContentsBitmapContext32(int width, int height, float scale) {
-    DisplayTexture* tex = NULL;
-
     if ([NSThread isMainThread]) {
-        tex = GetCACompositor()->CreateWritableBitmapTexture32(width, height);
-    }
+        DisplayTexture* tex = GetCACompositor()->CreateWritableBitmapTexture32(width, height);
+        RETURN_NULL_IF(!tex);
+
+        std::unique_ptr<IDisplayTextureImpl> displayTexture(new IDisplayTextureImpl(tex));
 
-    CGContextRef ret = _CGBitmapContextCreateWithTexture(width, height, scale, tex, &_globallockingBufferInterface);
+		Microsoft::WRL::ComPtr<IWICBitmap> bitmap = Microsoft::WRL::Make<CGIWICBitmap>(displayTexture.get(), GUID_WICPixelFormat32bppPBGRA, height, width);
+        //We want to convert it to GUID_WICPixelFormat32bppPBGRA for D2D Render Target compatibility.
+        CGImageRef texture = _CGImageCreateWithWICBitmap(bitmap.Get()); // +1 is taken care by CGContext
 
-    if (tex) {
-        _globallockingBufferInterface.ReleaseDisplayTexture(tex);
+        Microsoft::WRL::ComPtr<ID2D1Factory> factory;
+        RETURN_NULL_IF_FAILED(_CGGetD2DFactory(&factory));
+
+        Microsoft::WRL::ComPtr<ID2D1RenderTarget> renderTarget;
+        RETURN_NULL_IF_FAILED(factory->CreateWicBitmapRenderTarget(bitmap.Get(), D2D1::RenderTargetProperties(), &renderTarget));
+        renderTarget->SetDpi(kWindowsDPI * scale, kWindowsDPI * scale);
+
+		displayTexture.release();
+        return _CGBitmapContextCreateWithRenderTarget(renderTarget.Get(),texture);
```

Comment:
I'm going with "image" ?

---
In reply to: [87103132](https://github.com/Microsoft/WinObjC/pull/1332#discussion_r87103132) [](ancestors = 87103132,87102962)